src/algebra/quandle.lean

```diff
@@ -117,6 +126,46 @@ localized "infixr (name := shelf_hom) ` →◃ `:25 := shelf_hom" in quandles
 
 open_locale quandles
 
+namespace unital_shelf
+
+open shelf
+open unital_shelf
+
+variables {S : Type*} [unital_shelf S]
+
+/--
+A monoid is *graphic* if, for all `x` and `y`, the *graphic identity*
+`(x * y) * x = x * y` holds.  For a unital shelf, this graphic
+identity holds.
+-/
+lemma graphic_identity {x y : S} : (x ◃ y) ◃ x = x ◃ y :=
+begin
+  have h : (x ◃ y) ◃ x = (x ◃ y) ◃ (x ◃ 1) := by rw act_one,
+  rw [h, ←shelf.self_distrib, act_one],
+end
+
+lemma idempotent {x : S} : (x ◃ x) = x :=
```

Comment:
Here's a suggestion for a mathlib-friendly name. Feel free to create a docstring with the word "idempotent."
```suggestion
lemma self_act_eq {x : S} : (x ◃ x) = x :=
```